### PR TITLE
Move CI to github

### PR DESCRIPTION
**Pull request by n10ty**

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+jobs:
+  run-tests:
+    name: Run unit tests
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-go@v3
+      with:
+        go-version: 1.18
```
Comment (soffokl) [inline .github/workflows/build_and_test.yml:16]: Maybe try to migrate into 1.20 too?

**Review by soffokl — APPROVED** [review]
Looks good

File: .github/workflows/build_and_test.yml (hunk 1 above)
Comment (n10ty) [reply .github/workflows/build_and_test.yml:16]: Definitely. In separate PR. This is my next change

File: e2e/broker.go [hunk 2]
```diff
@@ -2,17 +2,18 @@ package e2e
 
 import (
 	"encoding/json"
-	"fmt"
-
 	v3 "github.com/mysteriumnetwork/discovery/proposal/v3"
```
Comment (yeszhanov95) [inline e2e/broker.go:5]: Node's using `v4`. Makes sense to use it as well

**Review by yeszhanov95 — APPROVED** [review]
In general, looks good to me. One suggestion: replace `v3` with `v4`. There's no big difference between then them, but it's better to stick with latest

File: e2e/broker.go (hunk 2 above)
Comment (n10ty) [reply e2e/broker.go:5]: Maybe. Not ready to switch now. In scope of other task

File: ci/e2e/e2e.go [hunk 3]
```diff
@@ -43,5 +43,9 @@ func upDependencies() error {
 }
 
 func upApp() error {
-	return sh.RunV("docker-compose", "-f", e2e.DockerFile, "up", "--build", "-d", "discovery")
+	err := sh.RunV("docker-compose", "-f", e2e.DockerFile, "up", "--build", "-d", "discovery")
```
Comment (Guillembonet) [inline ci/e2e/e2e.go:46]: any reason to not do this in the same command?

Comment (n10ty) [reply ci/e2e/e2e.go:46]: No. Didn't know it accept multiple services

Comment (Guillembonet) [reply ci/e2e/e2e.go:46]: I think it does

Comment (n10ty) [reply ci/e2e/e2e.go:46]: Correct. I checked. Will do this change in next pr

**Pull request merged**